projects/sel4/porting.md

```diff
@@ -7,79 +7,159 @@ SPDX-FileCopyrightText: 2020 seL4 Project a Series of LF Projects, LLC.
+the device tree blob (DTB) from building the Linux kernel to get device tree that
+Linux uses.
 
-Look for DTS and drivers inside the [Linux repository](https://github.com/torvalds/linux), normally they will be supported. If they are supported clone Linux and modify the `update-dts.sh` to add your target. Run the script located in `kernel/tools/dts/update-dts.sh` at the cloned Linux repository. This will create a "platform.dts" file from Linux inside the dts folder in the kernel. You may need to add device nodes depending on what Linux uses and what is required by seL4. For example, in the port of the Rockpro64 a memory node and an extra timer node were needed so these had to be defined manually.
+You can compile the Linux Device Tree Sources into a final Device Tree Blob with:
+```sh
+make ARCH=<arm/arm64/riscv> CROSS_COMPILE=<TOOLCHAIN> defconfig
+make ARCH=<arm/arm64/riscv> CROSS_COMPILE=<TOOLCHAIN> dtbs
+```
+
+The DTBs can be found in `arch/<ARCH>/boots/dts/`.
+
+The following command allows you to decompile the DTB:
+```sh
+dtc -I dtb -O dts /path/to/dtb -o linux.dts
+```
+
+It should be noted that you may need to add device nodes depending on what Linux uses
+and what is required by seL4. For example, in the port of the ROCKPro64 a memory node
+and an extra timer node were needed so these had to be defined manually.
 
 ### Hardware generation script
 
-Depending on your platform and the way the DTS is defined you may need to modify the hardware_gen.py script.
-From the above example the Rockpro64 needed a memory node so that this script could define working memory.
-Other gotchas could include interrupt cells, interrupt numbers being mapped wrongly, or incorrect addresses.
-Also serial and timer drivers will need to be added to the kernel if they don't already exist. You can check this through
-the DTS device strings and compare to the hardware.yml file in `kernel/tools/hardware.yml`. If needed add the required drivers in `kernel/drivers/`.
+Depending on your platform and the way the DTS is defined you may need to modify the
+`seL4/tools/hardware_gen.py` script.
+
+From the above example the ROCKPro64 needed a memory node so that this script could
+define working memory. Other gotchas could include interrupt cells, interrupt numbers being
+mapped wrongly, or incorrect addresses. Also serial and timer drivers will need to be added
+to the kernel if they don't already exist. You can check this through the DTS device strings
+and compare to the `seL4/tools/hardware.yml` file.
+
+If needed, add the required drivers in `seL4/src/drivers/` and `seL4/include/drivers/`.
 
 ### Kernel
 
-You will need to add `libsel4/sel4_plat_include/<platform>/sel4/plat/api/constants.h` and add sel4_UserTop depending on what you support. Look in the other platform files for examples similar to your platform.
+You will need to add `seL4/libsel4/sel4_plat_include/<platform>/sel4/plat/api/constants.h`,
```

Comment:
I suggest replacing the comma by a semi-colon, or full-stop.

Reply:
Done.